exec/cnex/cnex.c

```diff
@@ -0,0 +1,1217 @@
+    Cell *addr = top(self->stack)->address; pop(self->stack);
+
+    if (!number_is_integer(index)) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(index), BID_ZERO);
+        return;
+    }
+    int64_t i = number_to_sint64(index);
+    if (i < 0) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(number_from_sint64(i)), BID_ZERO);
+        return;
+    }
+    uint64_t j = (uint64_t)i;
+    push(self->stack, cell_fromAddress(cell_arrayIndexForWrite(addr, j)));
+}
+
+void exec_INDEXAV(TExecutor *self)
+{
+    self->ip++;
+    Number index = top(self->stack)->number; pop(self->stack);
+    Cell *array = top(self->stack);
+
+    if (!number_is_integer(index)) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(index), BID_ZERO);
+        return;
+    }
+    int64_t i = number_to_sint64(index);
+    if (i < 0) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(number_from_sint64(i)), BID_ZERO);
+        return;
+    }
+    uint64_t j = (uint64_t)i;
+    if (j >= array->array_size) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(number_from_uint64(j)), BID_ZERO);
+        return;
+    }
+    assert(j < array->array_size);
+    Cell *val = cell_fromCell(&array->array[j]);
+    pop(self->stack);
+    push(self->stack, val);
+}
+
+void exec_INDEXAN(TExecutor *self)
+{
+    self->ip++;
+    Number index = top(self->stack)->number; pop(self->stack);
+    Cell *array = top(self->stack);
+
+    if (!number_is_integer(index)) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(index), BID_ZERO);
+        return;
+    }
+    int64_t i = number_to_sint64(index);
+    if (i < 0) {
+        self->rtl_raise(self, "ArrayIndexException", number_to_string(number_from_sint64(i)), BID_ZERO);
+        return;
+    }
+    uint64_t j = (uint64_t)i;
+    /* ToDo: Debug this.  I think is following code is WRONG! */
+    Cell *val = (j < array->array_size ? cell_fromCell(&array->array[j]) : cell_newCell());
+    pop(self->stack);
+    push(self->stack, val);
+}
+
+void exec_INDEXDR()
+{
+    fatal_error("exec_INDEXDR not implemented");
+}
+
+void exec_INDEXDW()
+{
+    fatal_error("exec_INDEXDW not implemented");
+}
+
+void exec_INDEXDV()
+{
+    fatal_error("exec_INDEXDV not implemented");
+}
+
+void exec_INA(TExecutor *self)
+{
+    self->ip++;
+    Cell *array = cell_fromCell(top(self->stack)); pop(self->stack);
+    Cell *val = cell_fromCell(top(self->stack)); pop(self->stack);
+
+    BOOL v = cell_arrayElementExists(array, val);
+    cell_freeCell(val);
+    cell_freeCell(array);
+
+    push(self->stack, cell_fromBoolean(v));
+}
+
+void exec_IND()
+{
+    fatal_error("exec_IND not implemented");
+}
+
+void exec_CALLP(TExecutor *self)
+{
+    self->ip++;
+    unsigned int val = exec_getOperand(self);
+    const char *func = self->object->strings[val]->data;
+
+    if (!global_callFunction(func, self)) {
```

Comment:
Changed so that global_callFunction() throws a fatal_error if the function isn't found.  However, I'm not too fond of this.  Originally, I intended to have multiple static arrays of functions, one would be simply for the intrinsic functions, and one would be for the globals, then the ones for the modules.  I was going to arrange them so that intrinsic functions would be checked first, then globals, and so on, using the return value.  This can be readdressed later.